</div>


<a href="https://www.bigdatauniversity.com/?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDA0101ESCoursera33499360-2022-01-01"><img src="https://s3-api.us-geo.objectstorage.softlayer.net/cf-courses-data/CognitiveClass/DA0101EN/Images/CCLog.png" width=300, align="center"></a>

<h1 align=center><font size = 5>Análisis de Datos con Python</font></h1>


<h1>Introducción</h1>
<h3>¡Bienvenido!</h3>

<p>
En esta sección aprenderá como acercarse a la adquisición de datos de varias formas y obtener información necesaria de un conjunto de datos. Al terminar este laboratorio podra cargar con éxito los datos al cuaderno de Jupyter y obtener aportaciones fundamentales mediante la libreria Pandas.
   </p>


<h2>Indice</h2>

<div class="alert alert-block alert-info" style="margin-top: 20px">
<ol>
    <li><a href="https://#data_acquisition">Obtención de Datos</a>
    <li><a href="https://#basic_insight">Características Básicas del Conjunto de Datos</a></li>
</ol>

Estimated Time Needed: <strong>10 min</strong>

</div>
<hr>


<h1 id="data_acquisition">Obtención de Datos</h1>
<p>
Existen varios formatos para un conjunto de datos, .csv, .json, .xlsx, etc. Los datos pueden ser almacenados en distintos lugares, ya sea localmente o en línea.<b>
En estas sección aprenderá a cargar un conjunto de datos en su cuaderno de Jupyter.</b>
En nuestro caso el conjunto de datos Automobile es de una fuente en línea en formato CSV (valores separados por coma). Usemos este conjunto como ejemplo para practicar la lectura de datos.

<ul>
    <li>fuente de datos: <a href="https://archive.ics.uci.edu/ml/machine-learning-databases/autos/imports-85.data?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDA0101ESCoursera33499360-2022-01-01" target="_blank">https://archive.ics.uci.edu/ml/machine-learning-databases/autos/imports-85.data</a></li>
    <li>tipo de datos: csv</li>
</ul>
La librería Pandas es una herramienta util que nos permite la lectura de varios conjuntos de datos en un dataframe; nuestra plataforma de cuaderno de Jupyter posee esta librería por defecto así que lo único que debemos hacer es importar Pandas sin necesidad de instalarlo.
</p>


In [ ]:
# importar librería pandas 
import pandas as pd

<h2>Lectura de Datos</h2>
<p>
Utilizamos la función <code>pandas.read_csv()</code> para leer el archivo csv. Entre los corchetes pondremos la ruta del archivo sin comillas, de esta forma Pandas leerá el archivo en el dataframe desde esa dirección. La ruta del archivo puede ser tanto una URL como una dirección local.<br>
Debido a que los datos no tienen cabeceras podemos añadir un argumento <code>headers = None</code> dentro del método <code>read_csv()</code> para que Pandas no establezca de forma automática la primera fila como cabecera.<br>
Puede además asignar el conjunto de datos a cualquier variable que haya creado.

</p>


Este conjunto de datos fue almacenado como un objeto en IBM Cloud, haz clic <a href="https://cloud.ibm.com/registration?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDA0101ESCoursera33499360-2022-01-01">AQUI</a> para almacenar gratuitamente.


In [2]:
# Importar la librería Pandas 
import pandas as pd

# Leer el archivo en línea desde la URL de arriba y asignarla a la variable "df"
other_path = "https://s3-api.us-geo.objectstorage.softlayer.net/cf-courses-data/CognitiveClass/DA0101EN/auto.csv"
df = pd.read_csv(other_path, header=None)

Después de leer el conjunto de datos podemos utilizar el método <code>dataframe.head(n)</code> para revisar las primeras n filas del dataframe; donde n es u entero. Al contrario de <code>dataframe.head(n)</code>, <code>dataframe.tail(n)</code> mostrará las n filas del final del dataframe.


In [3]:
# muestra las primeras 5 filas mediante el método dataframe.head()
print("The first 5 rows of the dataframe") 
df.head(5)

The first 5 rows of the dataframe


0    1            2    3    4     5            6    7      8     9   ...  \
0   3    ?  alfa-romero  gas  std   two  convertible  rwd  front  88.6  ...   
1   3    ?  alfa-romero  gas  std   two  convertible  rwd  front  88.6  ...   
2   1    ?  alfa-romero  gas  std   two    hatchback  rwd  front  94.5  ...   
3   2  164         audi  gas  std  four        sedan  fwd  front  99.8  ...   
4   2  164         audi  gas  std  four        sedan  4wd  front  99.4  ...   

    16    17    18    19    20   21    22  23  24     25  
0  130  mpfi  3.47  2.68   9.0  111  5000  21  27  13495  
1  130  mpfi  3.47  2.68   9.0  111  5000  21  27  16500  
2  152  mpfi  2.68  3.47   9.0  154  5000  19  26  16500  
3  109  mpfi  3.19  3.40  10.0  102  5500  24  30  13950  
4  136  mpfi  3.19  3.40   8.0  115  5500  18  22  17450  

[5 rows x 26 columns]

<div class="alert alert-danger alertdanger" style="margin-top: 20px">
<h1> Pregunta #1: </h1>
<b>Revisar las ultimas 10 filas del dataframe "df"</b>    
</div>


In [5]:
# Escribe tu código aquí y presiona Shift+Enter para ejecutar
df.tail(10)

0    1      2       3      4     5      6    7      8      9   ...   16  \
195  -1   74  volvo     gas    std  four  wagon  rwd  front  104.3  ...  141   
196  -2  103  volvo     gas    std  four  sedan  rwd  front  104.3  ...  141   
197  -1   74  volvo     gas    std  four  wagon  rwd  front  104.3  ...  141   
198  -2  103  volvo     gas  turbo  four  sedan  rwd  front  104.3  ...  130   
199  -1   74  volvo     gas  turbo  four  wagon  rwd  front  104.3  ...  130   
200  -1   95  volvo     gas    std  four  sedan  rwd  front  109.1  ...  141   
201  -1   95  volvo     gas  turbo  four  sedan  rwd  front  109.1  ...  141   
202  -1   95  volvo     gas    std  four  sedan  rwd  front  109.1  ...  173   
203  -1   95  volvo  diesel  turbo  four  sedan  rwd  front  109.1  ...  145   
204  -1   95  volvo     gas  turbo  four  sedan  rwd  front  109.1  ...  141   

       17    18    19    20   21    22  23  24     25  
195  mpfi  3.78  3.15   9.5  114  5400  23  28  13415  
196  mpfi  3.78  3.15   9.5  114  5400  24  28  15985  
197  mpfi  3.78  3.15   9.5  114  5400  24  28  16515  
198  mpfi  3.62  3.15   7.5  162  5100  17  22  18420  
199  mpfi  3.62  3.15   7.5  162  5100  17  22  18950  
200  mpfi  3.78  3.15   9.5  114  5400  23  28  16845  
201  mpfi  3.78  3.15   8.7  160  5300  19  25  19045  
202  mpfi  3.58  2.87   8.8  134  5500  18  23  21485  
203   idi  3.01  3.40  23.0  106  4800  26  27  22470  
204  mpfi  3.78  3.15   9.5  114  5400  19  25  22625  

[10 rows x 26 columns]

<div class="alert alert-danger alertdanger" style="margin-top: 20px">
<h1> Pregunta #1 Respuesta: </h1>
<b>Ejecute el código de abajo para ver la solución</b>    
</div>


Haz doble clic <b>aquí</b> para ver la solución.

<!-- The answer is below:

print("The last 10 rows of the dataframe\n")
df.tail(10)

-->


<h3>Añadir Cabeceras</h3>
<p>
Observe el conjunto de datos; Pandas automaticamente establece la cabecera en un entero a partir de 0.  
</p>
<p>
Para describir mejor nuestros datos podemos agregarle una cabecera, esta información esta disponible en:    
<a href="https://archive.ics.uci.edu/ml/datasets/Automobile?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDA0101ESCoursera33499360-2022-01-01" target="_blank">https://archive.ics.uci.edu/ml/datasets/Automobile</a>
</p>
<p>
De este modo debemos agregar las cabeceras manualmente.    
</p>
<p>
Primero creamos una lista "headers" que incluya todos los nombres de columna en orden.
Despues usamos <code>dataframe.columns = headers</code> para reemplazar las cabeceras por la lista que hemos creado.
</p>


In [6]:
# crear la lista headers 
headers = ["symboling","normalized-losses","make","fuel-type","aspiration", "num-of-doors","body-style",
         "drive-wheels","engine-location","wheel-base", "length","width","height","curb-weight","engine-type",
         "num-of-cylinders", "engine-size","fuel-system","bore","stroke","compression-ratio","horsepower",
         "peak-rpm","city-mpg","highway-mpg","price"]
print("headers\n", headers)

headers
 ['symboling', 'normalized-losses', 'make', 'fuel-type', 'aspiration', 'num-of-doors', 'body-style', 'drive-wheels', 'engine-location', 'wheel-base', 'length', 'width', 'height', 'curb-weight', 'engine-type', 'num-of-cylinders', 'engine-size', 'fuel-system', 'bore', 'stroke', 'compression-ratio', 'horsepower', 'peak-rpm', 'city-mpg', 'highway-mpg', 'price']


Remplazamos las cabeceras y volvemos a revisar nuestro dataframe


In [7]:
df.columns = headers
df.head(10)

symboling normalized-losses         make fuel-type aspiration num-of-doors  \
0          3                 ?  alfa-romero       gas        std          two   
1          3                 ?  alfa-romero       gas        std          two   
2          1                 ?  alfa-romero       gas        std          two   
3          2               164         audi       gas        std         four   
4          2               164         audi       gas        std         four   
5          2                 ?         audi       gas        std          two   
6          1               158         audi       gas        std         four   
7          1                 ?         audi       gas        std         four   
8          1               158         audi       gas      turbo         four   
9          0                 ?         audi       gas      turbo          two   

    body-style drive-wheels engine-location  wheel-base  ...  engine-size  \
0  convertible          rwd           front        88.6  ...          130   
1  convertible          rwd           front        88.6  ...          130   
2    hatchback          rwd           front        94.5  ...          152   
3        sedan          fwd           front        99.8  ...          109   
4        sedan          4wd           front        99.4  ...          136   
5        sedan          fwd           front        99.8  ...          136   
6        sedan          fwd           front       105.8  ...          136   
7        wagon          fwd           front       105.8  ...          136   
8        sedan          fwd           front       105.8  ...          131   
9    hatchback          4wd           front        99.5  ...          131   

   fuel-system  bore  stroke compression-ratio horsepower  peak-rpm city-mpg  \
0         mpfi  3.47    2.68               9.0        111      5000       21   
1         mpfi  3.47    2.68               9.0        111      5000       21   
2         mpfi  2.68    3.47               9.0        154      5000       19   
3         mpfi  3.19    3.40              10.0        102      5500       24   
4         mpfi  3.19    3.40               8.0        115      5500       18   
5         mpfi  3.19    3.40               8.5        110      5500       19   
6         mpfi  3.19    3.40               8.5        110      5500       19   
7         mpfi  3.19    3.40               8.5        110      5500       19   
8         mpfi  3.13    3.40               8.3        140      5500       17   
9         mpfi  3.13    3.40               7.0        160      5500       16   

  highway-mpg  price  
0          27  13495  
1          27  16500  
2          26  16500  
3          30  13950  
4          22  17450  
5          25  15250  
6          25  17710  
7          25  18920  
8          20  23875  
9          22      ?  

[10 rows x 26 columns]

Podemos eliminar valores faltantes en la columna "price" de la siguiente manera


In [8]:
df.dropna(subset=["price"], axis=0)

symboling normalized-losses         make fuel-type aspiration  \
0            3                 ?  alfa-romero       gas        std   
1            3                 ?  alfa-romero       gas        std   
2            1                 ?  alfa-romero       gas        std   
3            2               164         audi       gas        std   
4            2               164         audi       gas        std   
..         ...               ...          ...       ...        ...   
200         -1                95        volvo       gas        std   
201         -1                95        volvo       gas      turbo   
202         -1                95        volvo       gas        std   
203         -1                95        volvo    diesel      turbo   
204         -1                95        volvo       gas      turbo   

    num-of-doors   body-style drive-wheels engine-location  wheel-base  ...  \
0            two  convertible          rwd           front        88.6  ...   
1            two  convertible          rwd           front        88.6  ...   
2            two    hatchback          rwd           front        94.5  ...   
3           four        sedan          fwd           front        99.8  ...   
4           four        sedan          4wd           front        99.4  ...   
..           ...          ...          ...             ...         ...  ...   
200         four        sedan          rwd           front       109.1  ...   
201         four        sedan          rwd           front       109.1  ...   
202         four        sedan          rwd           front       109.1  ...   
203         four        sedan          rwd           front       109.1  ...   
204         four        sedan          rwd           front       109.1  ...   

     engine-size  fuel-system  bore  stroke compression-ratio horsepower  \
0            130         mpfi  3.47    2.68               9.0        111   
1            130         mpfi  3.47    2.68               9.0        111   
2            152         mpfi  2.68    3.47               9.0        154   
3            109         mpfi  3.19    3.40              10.0        102   
4            136         mpfi  3.19    3.40               8.0        115   
..           ...          ...   ...     ...               ...        ...   
200          141         mpfi  3.78    3.15               9.5        114   
201          141         mpfi  3.78    3.15               8.7        160   
202          173         mpfi  3.58    2.87               8.8        134   
203          145          idi  3.01    3.40              23.0        106   
204          141         mpfi  3.78    3.15               9.5        114   

     peak-rpm city-mpg highway-mpg  price  
0        5000       21          27  13495  
1        5000       21          27  16500  
2        5000       19          26  16500  
3        5500       24          30  13950  
4        5500       18          22  17450  
..        ...      ...         ...    ...  
200      5400       23          28  16845  
201      5300       19          25  19045  
202      5500       18          23  21485  
203      4800       26          27  22470  
204      5400       19          25  22625  

[205 rows x 26 columns]

Ahora hemos leido con éxito el conjunto de datos y añadido las cabeceras correctas en el dataframe.


 <div class="alert alert-danger alertdanger" style="margin-top: 20px">
<h1> Pregunta #2: </h1>
<b>Encuentre el nombre de las columnas del dataframe</b>
</div>


In [10]:
# Escribe tu código aquí y presiona Shift+Enter para ejecutar
print(df.columns)


Index(['symboling', 'normalized-losses', 'make', 'fuel-type', 'aspiration',
       'num-of-doors', 'body-style', 'drive-wheels', 'engine-location',
       'wheel-base', 'length', 'width', 'height', 'curb-weight', 'engine-type',
       'num-of-cylinders', 'engine-size', 'fuel-system', 'bore', 'stroke',
       'compression-ratio', 'horsepower', 'peak-rpm', 'city-mpg',
       'highway-mpg', 'price'],
      dtype='object')


Haz doble clic <b>aquí</b> para ver la solución.

<!-- The answer is below:

print(df.columns)

-->


<h2>Guardar el Conjunto de Datos</h2>
<p>
De la misma forma, Pandas nos permite guardar el conjunto en formato CSV con el método <code>dataframe.to_csv()</code>, puede añadir la ruta al archivo y el nombre con comillas dentro de los corchetes.
</p>
<p>
    Por ejemplo, si guarda el dataframe <b>df</b> como <b>automobile.csv</b> en su equipo local, podría usar la sintaxis siguiente:
    </p>


Podemos leer y guardar con otros formatos y usar funciones similares a **`pd.read_csv()`** y **`df.to_csv()`** para otros formatos de datos, las funciones se muestran en la siguiente tabla:


<h2>Leer/Guardar en otros formatos</h2>

| Data Formate |        Read       |            Save |
| ------------ | :---------------: | --------------: |
| csv          |  `pd.read_csv()`  |   `df.to_csv()` |
| json         |  `pd.read_json()` |  `df.to_json()` |
| excel        | `pd.read_excel()` | `df.to_excel()` |
| hdf          |  `pd.read_hdf()`  |   `df.to_hdf()` |
| sql          |  `pd.read_sql()`  |   `df.to_sql()` |
| ...          |        ...        |             ... |


<h1 id="basic_insight">Características Básicas del Conjunto de Datos</h1>
<p>
Después de leer los datos en el dataframe Pandas es tiempo de explorar el conjunto de datos.<br>
Hay varias formas de obtener información esencial de los datos para entender mejor la totalidad de nuestro conjunto.</p>


<h2>Tipo de Datos</h2>
<p>
Los datos se encuentran en una variedad de tipos.<br>
Los tipos principales almacenados en dataframes de Pandas son <b>object</b>, <b>float</b>, <b>int</b>, <b>bool</b> y <b>datetime64</b>. Para aprender mejor acerca de cada atributo es mejor para nosotros saber el tipo de dato de cada columna. En Pandas:
</p>


In [11]:
df.dtypes


symboling              int64
normalized-losses     object
make                  object
fuel-type             object
aspiration            object
num-of-doors          object
body-style            object
drive-wheels          object
engine-location       object
wheel-base           float64
length               float64
width                float64
height               float64
curb-weight            int64
engine-type           object
num-of-cylinders      object
engine-size            int64
fuel-system           object
bore                  object
stroke                object
compression-ratio    float64
horsepower            object
peak-rpm              object
city-mpg               int64
highway-mpg            int64
price                 object
dtype: object

Regresa un seri con el tipo de dato de cada columna.


In [12]:
# revisar el tipo de dato del dataframe "df" con .dtypes
print(df.dtypes)

symboling              int64
normalized-losses     object
make                  object
fuel-type             object
aspiration            object
num-of-doors          object
body-style            object
drive-wheels          object
engine-location       object
wheel-base           float64
length               float64
width                float64
height               float64
curb-weight            int64
engine-type           object
num-of-cylinders      object
engine-size            int64
fuel-system           object
bore                  object
stroke                object
compression-ratio    float64
horsepower            object
peak-rpm              object
city-mpg               int64
highway-mpg            int64
price                 object
dtype: object


<p>
Como se muestra, se observa claramente que el tipo de dato de "symboling" y "curb-weight" es <code>int64</code>, "normalized-losses" es <code>object</code>, y "wheel-base" es <code>float64</code>, etc.
</p>
<p>
Estos tipos de datos pueden modificarse; aprenderemos a hacerlo en el siguiente módulo.    

</p>


<h2>Describe (método)</h2>
Si queremos obtener un resumen estadístico de cada columna, como el conteo, el valor medio de la columna, desviación estandar, etc. Utilizaremos el método describe:


Este método ofrece varios resumenes estadistícos, excluyendo valores  <code>NaN</code> (Not a Number).


In [13]:
df.describe()

symboling  wheel-base      length       width      height  \
count  205.000000  205.000000  205.000000  205.000000  205.000000   
mean     0.834146   98.756585  174.049268   65.907805   53.724878   
std      1.245307    6.021776   12.337289    2.145204    2.443522   
min     -2.000000   86.600000  141.100000   60.300000   47.800000   
25%      0.000000   94.500000  166.300000   64.100000   52.000000   
50%      1.000000   97.000000  173.200000   65.500000   54.100000   
75%      2.000000  102.400000  183.100000   66.900000   55.500000   
max      3.000000  120.900000  208.100000   72.300000   59.800000   

       curb-weight  engine-size  compression-ratio    city-mpg  highway-mpg  
count   205.000000   205.000000         205.000000  205.000000   205.000000  
mean   2555.565854   126.907317          10.142537   25.219512    30.751220  
std     520.680204    41.642693           3.972040    6.542142     6.886443  
min    1488.000000    61.000000           7.000000   13.000000    16.000000  
25%    2145.000000    97.000000           8.600000   19.000000    25.000000  
50%    2414.000000   120.000000           9.000000   24.000000    30.000000  
75%    2935.000000   141.000000           9.400000   30.000000    34.000000  
max    4066.000000   326.000000          23.000000   49.000000    54.000000

<p>
Este muestre el resumén estadístico de todas las columnas de tipo numerico (int, float).<br>
Por ejemplo, el atributo "symboling" tiene 205 recuentos, el valor medio de esta columna es 0.83, la desviación estandar es 1.25, el valor mínimo es -2, 25 percentil igual a 0, 50 percentil es 1, 75 percentil es 2 y el valor máximo es 3.
<br>
Sin embargo, si quisieramos además revisar todas las columnas incluyendo aquellas que son de tipo objeto, ¿Que tendríamos que hacer?  
<br><br>

Puede añadir un argumento <code>include = "all"</code> dentro del corchete. Hagámoslo de nuevo.

</p>


In [14]:
# describe todas las columnas de "df" 
df.describe(include = "all")

symboling normalized-losses    make fuel-type aspiration  \
count   205.000000               205     205       205        205   
unique         NaN                52      22         2          2   
top            NaN                 ?  toyota       gas        std   
freq           NaN                41      32       185        168   
mean      0.834146               NaN     NaN       NaN        NaN   
std       1.245307               NaN     NaN       NaN        NaN   
min      -2.000000               NaN     NaN       NaN        NaN   
25%       0.000000               NaN     NaN       NaN        NaN   
50%       1.000000               NaN     NaN       NaN        NaN   
75%       2.000000               NaN     NaN       NaN        NaN   
max       3.000000               NaN     NaN       NaN        NaN   

       num-of-doors body-style drive-wheels engine-location  wheel-base  ...  \
count           205        205          205             205  205.000000  ...   
unique            3          5            3               2         NaN  ...   
top            four      sedan          fwd           front         NaN  ...   
freq            114         96          120             202         NaN  ...   
mean            NaN        NaN          NaN             NaN   98.756585  ...   
std             NaN        NaN          NaN             NaN    6.021776  ...   
min             NaN        NaN          NaN             NaN   86.600000  ...   
25%             NaN        NaN          NaN             NaN   94.500000  ...   
50%             NaN        NaN          NaN             NaN   97.000000  ...   
75%             NaN        NaN          NaN             NaN  102.400000  ...   
max             NaN        NaN          NaN             NaN  120.900000  ...   

        engine-size  fuel-system  bore  stroke compression-ratio horsepower  \
count    205.000000          205   205     205        205.000000        205   
unique          NaN            8    39      37               NaN         60   
top             NaN         mpfi  3.62    3.40               NaN         68   
freq            NaN           94    23      20               NaN         19   
mean     126.907317          NaN   NaN     NaN         10.142537        NaN   
std       41.642693          NaN   NaN     NaN          3.972040        NaN   
min       61.000000          NaN   NaN     NaN          7.000000        NaN   
25%       97.000000          NaN   NaN     NaN          8.600000        NaN   
50%      120.000000          NaN   NaN     NaN          9.000000        NaN   
75%      141.000000          NaN   NaN     NaN          9.400000        NaN   
max      326.000000          NaN   NaN     NaN         23.000000        NaN   

        peak-rpm    city-mpg highway-mpg price  
count        205  205.000000  205.000000   205  
unique        24         NaN         NaN   187  
top         5500         NaN         NaN     ?  
freq          37         NaN         NaN     4  
mean         NaN   25.219512   30.751220   NaN  
std          NaN    6.542142    6.886443   NaN  
min          NaN   13.000000   16.000000   NaN  
25%          NaN   19.000000   25.000000   NaN  
50%          NaN   24.000000   30.000000   NaN  
75%          NaN   30.000000   34.000000   NaN  
max          NaN   49.000000   54.000000   NaN  

[11 rows x 26 columns]

<p>
Ahora se da el resumen estadistíco de todas las columnas, incluyendo los atributos de tipo objeto.<br>
Podemos observar ahora cuantos valores únicos, el cual es el valor mas alto y la frecuencia del valor alto en las columnas de tipo objeto.<br>
Algunos valores en la tabla de arriba se muestran como "NaN", esto se debe a que aquellos números no estan disponibles a pesar del tipo particular de columna.<br>    
</p>


<div class="alert alert-danger alertdanger" style="margin-top: 20px">
<h1> Pregunta #3: </h1>

<p>
Puede seleccionar las columna de un dataframe indicando el nombre de cada una, por ejemplo, puede seleccionar tres columnas de la siguiente manera:
</p>
<p>
    <code>dataframe[[' column 1 ',column 2', 'column 3']]</code>
</p>
<p>
Donde "column" es el nombre de la columna se puede aplicar el método ".describe()" para obtener las estadísticas de aquellas columnas de la siguiente manera:
</p>
<p>
    <code>dataframe[[' column 1 ',column 2', 'column 3'] ].describe()</code>
</p>
Aplicar el método ".describe()" a las columnas 'length' y 'compression-ratio'. 
</div>


In [17]:
# Escribe tu código aquí y presiona Shift+Enter para ejecutar
df[['width','height']].describe()

width      height
count  205.000000  205.000000
mean    65.907805   53.724878
std      2.145204    2.443522
min     60.300000   47.800000
25%     64.100000   52.000000
50%     65.500000   54.100000
75%     66.900000   55.500000
max     72.300000   59.800000

Haz doble clic <b>aquí</b> para ver la solución.

<!-- The answer is below:

df[['length', 'compression-ratio']].describe()

-->


<h2>Info (método)</h2>
Otro método que puede usar para revisar el dataframe es:


Ofrece un resumen conciso del dataframe.


In [18]:
# mirar la información de "df"
df.info

<bound method DataFrame.info of      symboling normalized-losses         make fuel-type aspiration  \
0            3                 ?  alfa-romero       gas        std   
1            3                 ?  alfa-romero       gas        std   
2            1                 ?  alfa-romero       gas        std   
3            2               164         audi       gas        std   
4            2               164         audi       gas        std   
..         ...               ...          ...       ...        ...   
200         -1                95        volvo       gas        std   
201         -1                95        volvo       gas      turbo   
202         -1                95        volvo       gas        std   
203         -1                95        volvo    diesel      turbo   
204         -1                95        volvo       gas      turbo   

    num-of-doors   body-style drive-wheels engine-location  wheel-base  ...  \
0            two  convertible          rwd      

<p>
Podemos observar la información del dataframe con las primeras 30 filas y las últimas 30.
<br><br>
Además, tambien muestra que la totalidad del dataframe tiene 205 filas y 26 columnas en total.
</p>


<h1>¡Excelente! ¡Ha completado el Cuaderno de Introducción!</h1>


<div class="alert alert-block alert-info" style="margin-top: 20px">

```
<p><a href="https://cocl.us/corsera_da0101en_notebook_bottom"><img src="https://s3-api.us-geo.objectstorage.softlayer.net/cf-courses-data/CognitiveClass/DA0101EN/Images/BottomAd.png" width="750" align="center"></a></p>
```

</div>


<h3>Acerca de los Autores:</h3>

Este cuaderno fue escrito por <a href="https://www.linkedin.com/in/mahdi-noorian-58219234/?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDA0101ESCoursera33499360-2022-01-01" target="_blank">Mahdi Noorian PhD</a>, <a href="https://www.linkedin.com/in/joseph-s-50398b136/?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDA0101ESCoursera33499360-2022-01-01" target="_blank">Joseph Santarcangelo</a>, Bahare Talayian, Eric Xiao, Steven Dong, Parizad, Hima Vsudevan y <a href="https://www.linkedin.com/in/fiorellawever/?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDA0101ESCoursera33499360-2022-01-01" target="_blank">Fiorella Wenver</a> y <a href="https:// https://www.linkedin.com/in/yi-leng-yao-84451275/ " target="_blank" >Yi Yao</a>.

<p><a href="https://www.linkedin.com/in/joseph-s-50398b136/?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDA0101ESCoursera33499360-2022-01-01" target="_blank">Joseph Santarcangelo</a> es un Cientifico de Datos en IBM que posee un Doctorado en Ingeniería Electrica. Su investigación se enfoca en el uso de Aprendizaje Automático, Procesamiento de Señales y Visión Artificial para determinar el impacto de los videos en la cognición. Joseph ha trabajado para IBM desde el término de su doctorado.
</p>


<hr>
<p>Copyright &copy; 2018 IBM Developer Skills Network. This notebook and its source code are released under the terms of the <a href="https://cognitiveclass.ai/mit-license/?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDA0101ESCoursera33499360-2022-01-01">MIT License</a>.</p>
